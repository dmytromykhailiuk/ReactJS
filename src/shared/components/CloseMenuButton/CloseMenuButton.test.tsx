import { shallow } from "enzyme";
import React from "react";
import { CloseMenuButton } from "./";
import toJson from 'enzyme-to-json';

describe("CheckboxIcon", () => {
  it("should match first snepshot", () => {
    const wrapper = shallow(<CloseMenuButton />)
    expect(toJson(wrapper)).toMatchSnapshot("first");
  })

  it("should match second snepshot", () => {
    const wrapper = shallow(<CloseMenuButton isSmall onCloseButtonClicked={() => {}} />)
    expect(toJson(wrapper)).toMatchSnapshot("second");
  })

  it("should call onClick when click on button", () => {
    const onClick = jest.fn();
    const wrapper = shallow(
      <CloseMenuButton 
        onCloseButtonClicked={onClick}
      >
        Button
      </CloseMenuButton>
    );
    wrapper.find("div div").simulate('click');

    expect(onClick).toHaveBeenCalled();
  })
})