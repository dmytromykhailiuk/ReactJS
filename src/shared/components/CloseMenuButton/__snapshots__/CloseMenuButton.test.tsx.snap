// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxIcon should match first snepshot: first 1`] = `
<div
  className="close-menu-button__wrapper"
>
  <div
    className="close-menu-button__button"
    onClick={[Function]}
  />
</div>
`;

exports[`CheckboxIcon should match second snepshot: second 1`] = `
<div
  className="close-menu-button__wrapper"
>
  <div
    className="close-menu-button__button close-menu-button__button--small"
    onClick={[Function]}
  />
</div>
`;
